### DISCUSSION TITLE
Hotreloading with Websocket

### DISCUSSION THREAD

**Discussion by cpietsch** [community/community#26911] [Codespaces]
<p>Hello, I am trying to get a hotreloading feature working which is using websocket on port 3000.<br>
Codespace is automatically forwarding port 80 to 3000 internally, but the script is still trying to poll 3000.<br>
Is there a way to open / forward port 3000 on the <a href="http://xxxx.apps.codespaces.githubusercontent.com" rel="nofollow noopener">xxxx.apps.codespaces.githubusercontent.com</a> url ?</p>
<p>More details here: <a href="https://github.com/vitejs/vite/issues/797" rel="nofollow noopener">https://github.com/vitejs/vite/issues/797</a></p>

**Comment by lostintangent** [comment]
<p>Hey! Thanks for reporting this. Out of curiosity: are you able to specify port 443 to the Vite CLI when starting it up? I’m just curious if that would provide a workaround in the meantime, while we look into how to best support this specific tool :+1:</p>

**Comment by lostintangent** [comment, marked as answer]
<p>Update: This issue is because the dev server is “forwarded” over port 443, but the Vite HMR client is being set to 3000. I just sent a <a href="https://github.com/vitejs/vite/pull/800" rel="nofollow noopener">PR to Vite</a> to fix this, and in the meantime, you should be able to workaround it by running <code>sudo vite --port 443</code> (<code>sudo</code> is required since 443 is a “protected” port). Let me know if that works for you!</p>

**Comment by cpietsch** [comment]
<p>Hey Jonathan,</p>
<p>That was fast! Thx a lot for looking into that issue and providing the fix.<br>
I really love codespace and can’t wait to use it extensively on what ever device I carry around.</p>
<p>Cheers Chris</p>

**Comment by subbuke** [comment]
it worked . Thanks man
